tests/adc_ng/main.c

```diff
@@ -0,0 +1,478 @@
+    print_str("V\n");
+
+    return 0;
+}
+
+static int cmd_vcc(int argc, char **argv)
+{
+    if (argc != 2) {
+        print_str("Usage: ");
+        print_str(argv[0]);
+        print_str(" <ADC>\n");
+        return -1;
+    }
+
+    uint8_t adc = atoi(argv[1]);
+    if (is_adc_invalid(adc)) return -1;
+
+    print_str("adc_ng_vcc(");
+    print_u32_dec(adc);
+    print_str(", &value)\n");
+
+    uint16_t value;
+    int retval = adc_ng_vcc(adc, &value);
+    if (retval) {
+        print_str("adc_ng_vcc() failed with ");
+        print_s32_dec(retval);
+        print_str("\n");
+        return -1;
+    }
+    char buf[16];
+    print_str("value = ");
+    print(buf, fmt_s32_dfp(buf, value, -3));
+    print_str("V\n");
+
+    return 0;
+}
+
+static int cmd_ntc(int argc, char **argv)
+{
+    if (argc != 2) {
+        print_str("Usage: ");
+        print_str(argv[0]);
+        print_str(" <ADC>\n");
+        return -1;
+    }
+
+    uint8_t adc = atoi(argv[1]);
+    if (is_adc_invalid(adc)) return -1;
+
+    print_str("adc_ng_internal_ntc(");
+    print_u32_dec(adc);
+    print_str(", &value)\n");
+
+    int16_t value;
+    int retval = adc_ng_internal_ntc(adc, &value);
+    if (retval) {
+        print_str("adc_ng_internal_ntc() failed with ");
+        print_s32_dec(retval);
+        print_str("\n");
+        return -1;
+    }
+    char buf[16];
+    print_str("value = ");
+    print(buf, fmt_s32_dfp(buf, value, -1));
+    print_str("°C\n");
+
+    return 0;
+}
+
+static int cmd_entropy(int argc, char **argv)
+{
+    if (argc != 3) {
+        print_str("Usage: ");
+        print_str(argv[0]);
+        print_str(" <ADC> <NUM_BYTES>\n");
+        return -1;
+    }
+
+    uint8_t adc = atoi(argv[1]);
+    if (is_adc_invalid(adc)) return -1;
+
+    size_t size = atoi(argv[2]);
+
+    if (size > sizeof(burst_buf)) {
+        print_str("Given size too big, must be at most ");
+        print_u32_dec(sizeof(burst_buf));
+        print_str("\n");
+        return -1;
+    }
+
+    print_str("adc_ng_entropy(");
+    print_u32_dec(adc);
+    print_str(", buf, ");
+    print_u32_dec(size);
+    print_str(")\n");
+
+    int retval = adc_ng_entropy(adc, burst_buf, size);
+    if (retval) {
+        print_str("adc_ng_internal_ntc() failed with ");
+        print_s32_dec(retval);
+        print_str("\n");
+        return -1;
+    }
+
+    uint8_t *buf = (uint8_t *)burst_buf;
+    print_str("Output:\n");
+    char printbuf[2];
+    fmt_byte_hex(printbuf, buf[0]);
+    print(printbuf, 2);
+    for (size_t i = 1; i < size; i++) {
+        print((i % 8) ? " " : "\n", 1);
+        fmt_byte_hex(printbuf, buf[i]);
+        print(printbuf, 2);
+    }
+    print_str("\n");
+
+    return 0;
+}
+
+static void print_driver(uint8_t adc)
+{
+    const adc_ng_driver_t *drv = adc_ng_drivers[adc];
+    print_str("Resolutions supported: ");
+    uint8_t lowest = adc_ng_min_res(adc);
+    print_u32_dec(lowest);
+    print_str("bit");
```

Comment:
```suggestion
    print_str(" bit");
```

Reply:
Fixed. (And other instances of this well, hopefully)